### DISCUSSION TITLE
Energy Based Model Implementation

### DISCUSSION THREAD

**Discussion by thhapke** [tracel-ai/burn#4715] [General]
I recently stumbled across a LeCun blog about using Energy Based Models as an alternative to the classical approach. 
- [http://yann.lecun.com/exdb/publis/pdf/lecun-06.pdf](http://yann.lecun.com/exdb/publis/pdf/lecun-06.pdf)
- [https://www.youtube.com/watch?v=BqgnnrojVBI](https://www.youtube.com/watch?v=BqgnnrojVBI)

Is there an initiative to implement the required functionality into BURN? 

Cheers Thorsten

**Comment by laggui** [comment]
Sorry for the late reply!

There is no planned work on EBMs. I think you should be able to implement it with a custom training loop, though iirc some optimization methods might require higher order gradients? Currently autodiff is only for first order.
